### PR TITLE
0.5 comments

### PR DESCRIPTION
**Pull request by zbraniecki**
Built on top of the private msg PR. Need to add tests, but hopefully a good start :)

### CODE REVIEW

File: src/syntax/parser.rs.orig [hunk 1]
```diff
@@ -0,0 +1,721 @@
+pub use super::errors::ParserError;
```
Comment (stasm) [inline src/syntax/parser.rs.orig]: Please remove this file.

File: src/syntax/parser.rs [hunk 2]
```diff
@@ -152,6 +143,11 @@ where
 {
     let id = get_private_identifier(ps)?;
 
+    if let Some(ast::Comment::Comment { .. }) = comment {
+    } else if comment.is_some() {
```
Comment (stasm) [inline src/syntax/parser.rs]: I don't understand this code. It looks like both `if let` and `else if` test if `comment` is a `Some`?

Comment (zbraniecki) [reply src/syntax/parser.rs]: heh, yeah, it's awkward. What I'm trying to test is if the variant of the Comment passed to get_message is actually `ast::Comment::Comment` and not GroupComment or ResourceComment.

If it's not, then I'm verifying that it's None.

Any ideas how to write it better? :)

Comment (stasm) [reply src/syntax/parser.rs]: Ahh, I see now, thanks. I might have wrongly suggested to use `if let` here then. Maybe a `match` like this?

```rust
match comment {
    Some(ast::Comment::GroupComment { .. }) => error!(),
    Some(ast::Comment::ResourceComment { .. }) => error!(),
    _ => ()
}
```

In the future, https://github.com/rust-lang/rfcs/issues/935 will help.

**Pull request merged**